walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -27,19 +27,43 @@
 * the Time Range object will represent that entire chunk of time rather than 4pm - 5pm, even if the duration of meeting is 1 hour. 
 */
 public final class FindMeetingQuery {
-  public static final int MIN_INCREMENT = 10;
+  public static final int MIN_INCREMENT = 5;
 
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
     long mtngDuration = request.getDuration();
-    Collection<String> attendees = request.getAttendees();
+    Collection<String> mandatoryAttendees = request.getAttendees();
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
     Collection<Event> allEvents = events;
 
+    Collection<TimeRange> meetingTimes = new ArrayList<TimeRange>();
+
+    if (optionalAttendees.size() != 0) {
+      Collection<String> allAttendees = new ArrayList<String>();
+      allAttendees.addAll(optionalAttendees);
+      allAttendees.addAll(mandatoryAttendees);
+    
+      HashMap<Integer, Integer> mandatoryEvents = makeMandatoryEventMap(allEvents, allAttendees);
+      meetingTimes = findMeetingTimes(mtngDuration, mandatoryEvents);
```

Comment:
Could this also use the same three-argument version of `findMeetingTimes` that the next two cases use?